syntax = "proto3";

package mail;

service MailService {
  rpc SendMail (MailRequest) returns (MailResponse) {}
  rpc CheckBounceStatus (BounceRequest) returns (BounceResponse) {}
}

message MailRequest {
  string from = 1;  
  string to = 2;
  string subject = 3;
  string body = 4;
  repeated Attachment attachments = 5; // Lista de archivos adjuntos
}

message Attachment {
  string filename = 1; // Nombre del archivo
  bytes content = 2;   // Contenido del archivo en formato binario
}

message MailResponse {
  string status = 1;
}

message BounceRequest {
  string from = 1;        // Correo del remitente
  string to = 2;          // Correo del destinatario
  string sent_time = 3;   // Hora de envío del correo (ISO 8601)
}

message BounceResponse {
  string status = 1;      // "failed", "completed", or "pending"
  string reason = 2;      // Razón del fallo (si aplica)
}